### PR TITLE
add switch to ignore default shortcuts at startup

### LINKED ISSUE #10452: deleted default shortcuts don't stay deleted
1. Open the shortcut tab of the preferences window and select global > panels in the top pane. 
2. Select global/panels/right (ctrl+shift+r) in the bottom pane and press the delete key. Dialog appears asking for confirmation. 
3. Click yes and the shortcut is removed.
4. Close and re-open darktable and repeat. Shortcut has been reinstated.

User created shortcuts are correctly removed.

### PR DESCRIPTION
**Pull request by dterrahe**
fixes #10452 as per https://github.com/darktable-org/darktable/issues/10452#issuecomment-1059524606

@elstoc please suggest better phrasing if needed or a different location in prefs

This PR also adds a confirmationless mapping mode. If you press the shortcut button while holding ctrl, you can overwrite existing shortcuts without a dialog popping up. DANGER, obviously. @AxelG-DE I thought you might like this as it makes it quite fast to switch between two different sets of mappings for midi knobs. Obviously you can set up a shortcut to the shortcut button with the "ctrl-toggle" _effect_ so you get this behavior by default.

### CODE REVIEW

File: data/darktableconfig.xml.in [hunk 1]
```diff
@@ -67,6 +67,13 @@
     <shortdescription>enable shortcut fallbacks</shortdescription>
     <longdescription>enables default meanings for additional buttons, modifiers or moves when used in combination with a base shortcut</longdescription>
   </dtconfig>
+  <dtconfig prefs="misc" section="interface">
+    <name>accel/load_defaults</name>
+    <type>bool</type>
+    <default>true</default>
+    <shortdescription>restore default shortcuts at startup</shortdescription>
+    <longdescription>load default shortcuts before user settings add/overwrite. switch off to prevent deleted defaults returning</longdescription>
```
Comment (elstoc) [inline data/darktableconfig.xml.in:75]: Perhaps would be fine without the words "add/overwrite".

File: data/darktableconfig.xml.in [hunk 2]
```diff
@@ -67,6 +67,13 @@
     <shortdescription>enable shortcut fallbacks</shortdescription>
     <longdescription>enables default meanings for additional buttons, modifiers or moves when used in combination with a base shortcut</longdescription>
   </dtconfig>
+  <dtconfig prefs="misc" section="interface">
+    <name>accel/load_defaults</name>
+    <type>bool</type>
+    <default>true</default>
+    <shortdescription>restore default shortcuts at startup</shortdescription>
```
Comment (elstoc) [inline data/darktableconfig.xml.in:74]: Perhaps "load" instead of "restore", since the latter kind of suggests user settings will be overwritten

**Comment by elstoc** [comment]
Of course, this does have the downside that the user won't get any newly-added defaults in future versions. I'm not sure there's a good way around that though. Disclaimer - I haven't tested this yet!

**Comment by dterrahe** [comment]
> Of course, this does have the downside that the user won't get any newly-added defaults in future versions.

As discussed; this is the reason it was originally set up as it was.

Let's hope that the Venn of people that want to delete default shortcuts without using those keys for something else and those that would hate getting a bunch of new shortcuts pushed down their throats is somewhat circular.

As said before, you can still (in the shortcuts dialog) _restore defaults_ (and then _restore edits_ in case you overwrite some of the defaults). Or if you want more control (and not have to delete all the previously deleted defaults again just to pick up one new one) you can use `diff ~/.config/darktable/shortcutsrc{.defaults,}`

> I'm not sure I understand this

1 Switch on shortcuts visual mapping mode by just clicking on the button
2 hover over a mappable widget
3 press a
4 hover over another mappable widget
5 press a again

switch off mapping mode and switch it on again, but now hold Ctrl while clicking the button.
do 1-5 again. notice the difference

**Comment by elstoc** [comment]
> but now hold Ctrl while clicking the button

Ah I was missing _where_ you had to press Ctrl. Understood now.

**Comment by AxelG-DE** [comment]
@dterrahe thanks for pinging me. I'll check this out as soon as I am not bound with house and taxation anymore (so May, if everything goes well)
So that indicates me, currently we still can use one layer of the midi board and maybe will never change?
Will be funny, what happens if I use a second or I also need to buy this xyz2000 thingie (I am quite handy caped with phone only atm)

**Comment by dterrahe** [comment]
> bound with house and taxation

Good luck with that!

> we still can use one layer of the midi board and maybe will never change?

Correct. You can assign different meanings to midi knobs depending on ctrl/shift/alt and you can define midi buttons to those modifiers and use the "toggle" effect (rather than "hold") so they stay switched on when pressed, but they will also impact mouse actions on masks etc (they'll act like a _real_ ctrl/...)

_IF_ there is real demand for more configurability, more thought (and agreement) is needed to expand. And the question whether that should live inside InputNG (so you could also have multiple sets of shortcuts for keyboard) or within midi. It could be as simple as just changing the knob/button number depending on which set is selected by a key. There could be an intermediate layer in lua (at the midi or inputng level) that intercepts any interactions and presents a virtual, fully customisable, view of the midi device to the system. Drawback of doing it at that level is that it needs to be done again for any other type of device (gamepads also suffer from having few joysticks). And there will not be any interaction between devices (switching layout/mode on midi leaves the mode on other devices unchanged).

Since InputNG is already criticised for being too complex, I'm currently not very motivated to take it a step further.

Having said that, on x-touch mini you _can_ use both layers. It's just that dt doesn't notice a switch until you touch some other knob/key. There _is_ a solution to that too: https://github.com/dterrahe/darktable/discussions/24

> what happens if I use a second

You can have two (up to 10) midi devices connected and their configurations are independent.

**Comment by AxelG-DE** [comment]
Thanks for your swift and comprehensive reply!

I agree, not to make things more complex.

8 kind of layers was already confusing. Layer A and B is good enough, if you ask me. That hack is cool but I will just slightly schift the anyway useless slider then :-)

Thanks for everything!!!

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**